packages/core/src/components/ImageOverlay.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useState } from 'react'
+import { ExamComponentProps } from '../createRenderChildNodes'
+import { getNumericAttribute, mapChildElements } from '../dom-utils'
+import { faEye } from '@fortawesome/free-solid-svg-icons/faEye'
+import { faEyeSlash } from '@fortawesome/free-solid-svg-icons/faEyeSlash'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ResponsiveMediaContainer from './ResponsiveMediaContainer'
+import { CommonExamContext } from './CommonExamContext'
+import classNames from 'classnames'
+
+function ImageOverlay(props: ExamComponentProps) {
+  const { element } = props
+  const [sliderValues, setSliderValue] = useState<number[]>(
+    Array.from({ length: element.children.length }, (_, i) => (i === 0 ? 1 : 0))
+  )
+
+  const imageElements = mapChildElements(element, (childElement) => childElement)
+
+  const onSetSliderValue = (event: React.ChangeEvent<HTMLInputElement>, index: number) => {
+    const updatedSliders = [...sliderValues]
+    updatedSliders[index] = Number(event.target.value)
+    setSliderValue(updatedSliders)
+  }
+
+  return (
+    <div className="e-image-overlay e-mrg-b-1">
+      <div className="e-columns e-columns--center-h">
+        <div>
+          {sliderValues.map((sliderValue, index) => (
+            <Slider
+              {...props}
+              key={index}
+              opacity={sliderValue}
+              element={imageElements[index]}
+              setSliderValue={(e) => onSetSliderValue(e, index)}
+            />
+          ))}
+        </div>
+      </div>
+      <div>
+        <div className="e-image-overlay__images-wrapper">
+          {sliderValues.map((slider, index) => {
+            if (!element.childNodes) return null
+            return (
+              <div
+                className="e-columns e-columns--center-h"
+                key={index}
+                style={{
+                  opacity: slider,
+                }}
+              >
+                <ImageOverlayImage
+                  {...props}
+                  className={classNames({ 'e-image-overlay__image-absolute': index > 0 })}
+                  element={imageElements[index]}
+                />
+              </div>
+            )
+          })}
+        </div>
+      </div>
+    </div>
+  )
+}
+
+interface SliderProps {
+  opacity: number
+  setSliderValue: (event: React.ChangeEvent<HTMLInputElement>) => void
```

Comment:
```suggestion
  setSliderValue: React.ChangeEventHandler<HTMLInputElement>
```

olis varmaan vielä vähän söpompi?